ogr/ogrsf_frmts/sqlite/ogrsqlitelayer.cpp

```diff
@@ -900,11 +900,18 @@ OGRFeature *OGRSQLiteLayer::GetNextRawFeature()
                 {
                     const GIntBig nVal =
                         sqlite3_column_int64(m_hStmt, iRawField);
-                    if (nVal >= INT_MIN && nVal <= INT_MAX)
-                        poFeature->SetFieldSameTypeUnsafe(
-                            iField, static_cast<int>(nVal));
+                    if (poFieldDefn->GetSubType() == OFSTBoolean)
```

Comment:
@rouault this change is only for consistency with other drivers but I'm not entirely sure , what do you think?

Reply:
Seems reasonable